python/ray/autoscaler/commands.py

```diff
@@ -262,129 +546,208 @@ def get_or_create_head_node(config, config_file, no_restart, restart_only, yes,
             head_node = None
 
         if not head_node:
-            confirm("This will create a new cluster", yes)
+            cli_logger.confirm(
+                yes,
+                "No head node found. "
+                "Launching a new cluster.", _abort=True)
+            cli_logger.old_confirm("This will create a new cluster", yes)
         elif not no_restart:
-            confirm("This will restart cluster services", yes)
+            cli_logger.old_confirm("This will restart cluster services", yes)
+
+        if head_node:
+            if restart_only:
+                cli_logger.confirm(
+                    yes,
+                    "Updating cluster configuration and "
+                    "restarting the cluster Ray runtime. "
+                    "Setup commands will not be run due to `{}`.\n",
+                    cf.bold("--restart-only"),
+                    _abort=True)
+            elif no_restart:
+                cli_logger.print(
+                    "Cluster Ray runtime will not be restarted due "
+                    "to `{}`.")
```

Comment:
this doesn't seem to have a fill-in string

Reply:
fixed! good catch